### PR TITLE
#1259 Rebase

### PR DESCRIPTION
**Pull request by pbauman**
Redux of #1259 with several fixes squashed in. This should address all comments from the #1259.

### CODE REVIEW

**Comment by pbauman** [comment]
Rebased on master while I was at it.

**Comment by pbauman** [comment]
990c73d in particular has the squashed in changes to override via command line and keeps around the input file. 2148606 has squashed in fixes for always using SPARSELU with any Eigen solver, not just in complex mode.

This should be good to merge, assuming CIVET is still happy.

File: src/systems/diff_system.C [hunk 1]
```diff
@@ -209,4 +192,174 @@ void DifferentiableSystem::release_linear_solver(LinearSolver<Number> *) const
...
+                                                    dbc.g_fem.get());
+
+                  else
+                    new_dbc = new DirichletBoundary(dbc.b,
+                                                    vars_to_add,
+                                                    *this,
+                                                    dbc.f_fem.get());
+                }
+              else
+                libmesh_error_msg("Could not find valid boundary function!");
```
Comment (roystgnr) [inline src/systems/diff_system.C]: Didn't we decide that this was wrong?  The best we can do for now is add ZeroFunction DBCs for the dot vars, or scream-and-die if the primal var DBCs are FEM-dependent or time-dependent?

Comment (pbauman) [inline src/systems/diff_system.C]: Yes, and it was fixed in 9edd8ba. Should I squash that fix commit in?

Comment (roystgnr) [inline src/systems/diff_system.C]: No, that's fine; I just missed it.

**Comment by roystgnr** [comment]
:+1: when Civet's done.

**Comment by jwpeterson** [comment]
After merging this, I get a compile error:
```
../src/systems/diff_system.C: In member function ‘bool libMesh::DifferentiableSystem::have_first_order_scalar_vars() const’:
../src/systems/diff_system.C:338:56: error: ‘FEFamily’ is not a class or namespace
           if( this->variable(*var_it).type().family == FEFamily::SCALAR )
                                                        ^
```
The CI was happy... is my compiler just dumb?

**Comment by jwpeterson** [comment]
Maybe it's a C++11 thing that our no-C++11 test doesn't catch.  I happen to be configured with `--disable-cxx11` on GCC 5.2.0 in this particular case...

**Comment by pbauman** [comment]
Thanks!

**Pull request merged**